bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs

```diff
@@ -89,19 +104,79 @@ where
 		fn prepare(message: Message) -> Result<PreparedMessage, ConvertMessageError> {
 			let mut remote_xcm: Xcm<()> = Xcm::new();
 
-			// Allow xcm decode failure so that assets can be trapped on AH instead of this
-			// message failing but funds are already locked on Ethereum.
-			if let Ok(versioned_xcm) = VersionedXcm::<()>::decode_with_depth_limit(
-				MAX_XCM_DECODE_DEPTH,
-				&mut message.xcm.as_ref(),
-			) {
-				if let Ok(decoded_xcm) = versioned_xcm.try_into() {
-					remote_xcm = decoded_xcm;
-				}
-			}
-
 			let ether_location = Location::new(2, [GlobalConsensus(EthereumNetwork::get())]);
 
+			match &message.xcm {
```

Comment:
I know this is a draft - just want to point out that there's way too many levels of nesting here.

Reply:
Refactored in https://github.com/vgeddes/polkadot-sdk/pull/13/commits/194ab7c9a4b932525b28975610a8a53d3100f06c.